src/main/java/com/laytonsmith/core/functions/Threading.java

```diff
@@ -586,4 +587,212 @@ public Version since() {
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
+			Thread th;
+			if(args.length == 1) {
+				th = THREAD_ID_MAP.get(args[0].val());
+				if(th != null) {
+					th.interrupt();
+				}
+			} else {
+				Thread.currentThread().interrupt();
+			}
+			return CVoid.VOID;
+		}
+
+		@Override
+		public Version since() {
+			return MSVersion.V3_3_4;
+		}
+
+		@Override
+		public String getName() {
+			return "x_interrupt";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{0, 1};
+		}
+
+		@Override
+		public String docs() {
+			return "void {[string id]} Interrupts the thread with the given id, or the current thread if no id is given."
+					+ " When a thread is interrupted, its interrupt status is set to true."
+					+ " This status can be checked using the " + new x_is_interrupted().getName() + " and " + new x_clear_interrupt().getName() + " function."
+					+ " Note that some blocking functions will throw an InterruptedException when the thread on which they are executed is interrupted.";
+		}
+	}
+
+	@api
+	@seealso({x_interrupt.class, x_clear_interrupt.class})
+	public static class x_is_interrupted extends AbstractFunction {
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return null;
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
+			if(args.length == 1) {
+				Thread th = THREAD_ID_MAP.get(args[0].val());
+				if(th != null) {
+					return CBoolean.get(th.isInterrupted());
+				} else {
+					return CBoolean.FALSE;
+				}
+			} else {
+				return CBoolean.get(Thread.currentThread().isInterrupted());
+			}
+		}
+
+		@Override
+		public Version since() {
+			return MSVersion.V3_3_4;
+		}
+
+		@Override
+		public String getName() {
+			return "x_is_interrupted";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{0, 1};
+		}
+
+		@Override
+		public String docs() {
+			return "boolean {[string id]} Tests whether this thread with the given id has been interrupted."
+					+ " If id not passed, it is be current thread."
+					+ " The \"interrupt status\" of the thread is unaffected by this method."
+					+ " In case if the thread does not exist or is not alive, return false";
+		}
+	}
+
+	@api
+	@seealso({x_interrupt.class, x_is_interrupted.class})
+	public static class x_clear_interrupt extends AbstractFunction {
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return null;
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
+			return CBoolean.get(Thread.interrupted());
```

Comment:
Just call interrupted here. No need to return the result, or expose the java weirdness here.

Reply:
@LadyCailin It could actually be useful to return that boolean. Having a clear function return whether it has actually cleared something or not allows one to in this case omit the `x_is_interrupted` call.

Reply:
Ok, fair enough, but then the docs need to change. The primary function of this function is to clear the interrupt status, in most cases, it should be good enough to simply call it and ignore the return - after this function is called, the thread's interrupt flag will be cleared, whether this call did it, or another one.